### ISSUE TITLE
Functions don't properly work with other window because of failing RegExp check

### ISSUE THREAD

**Issue by JiriTrecak** [blakeembrey/change-case#302]
Hello, I have been using the library in a pretty complex setup to build a library that will be distributed and I have started to try it out in our system and found a problem that prevents the usage of the library altogether.

When the library is compiled through a webpack, somewhere in the code, the following will occur (which I think is in no-case, the reduction):

```typescript
          return r instanceof RegExp
            ? e.replace(r, t)
            : r.reduce(function (e, r) {
                return e.replace(r, t)
              }, e)
```

I have tracked the issue to the specific line - this code will work normally as a primary dependency, but if this built library is then used in other contexts, for example, running through a Node VM as an executable or used in different spawned window, it will fail. The reason for it is:

```typescript
console.log(r instanceof RegExp) // false
console.log(r.constructor.name) // "RegExp"
```

Basically, what is happening is explained here:
https://medium.com/@samjwright/why-is-instanceof-my-object-returning-false-in-typescript-fec74df5c2a8

The fix is very simple, check for RegExp instance in another way.

Happy to open PR that fixes this should you be interested. I am of course aware that this is a highly exotic scenario, but it does fail this way (not everyone runs the code through NodeVM), so I think it should be fixed anyway.

Thank you very much!

**Comment by blakeembrey** [comment]
Thanks for the flag! I've actually been working on a small rewrite this week of the library that would coincidentally fix this issue. I just pushed my work so far up to GitHub, but I'm going to wait for a chunk of time to release/deprecate everything according to the updates. So I should have a fix for you available either tonight or tomorrow.

**Comment by blakeembrey** [comment]
This has been fixed with `change-case@5.0.0`. Please update and let me know if you still see any issues.
